package/src/ssr/WrappedApolloClient.test.tsx

```diff
@@ -0,0 +1,201 @@
+      link,
+    });
+
+    let finishedRenderCount = 0;
+
+    function Child() {
+      const { data } = useSuspenseQuery(QUERY_ME);
+      finishedRenderCount++;
+      return <>{data.me}</>;
+    }
+
+    const { findByText } = render(
+      <Provider makeClient={() => client}>
+        <Suspense fallback={"Fallback"}>
+          <Child />
+        </Suspense>
+      </Provider>
+    );
+
+    assert.deepStrictEqual(startedRequests, [FIRST_REQUEST]);
+    assert.deepStrictEqual(requestData, []);
+    assert.deepStrictEqual(staticData, []);
+
+    link.simulateResult({ result: { data: FIRST_RESULT } }, true);
+
+    await findByText("User");
+
+    assert.deepStrictEqual(requestData, [FIRST_WRITE]);
+    assert.deepStrictEqual(startedRequests, [FIRST_REQUEST]);
+    assert.deepStrictEqual(
+      staticData,
+      new Array(finishedRenderCount).fill(FIRST_HOOK_RESULT)
+    );
+  }
+);
+
+await testIn("browser")(
+  "`useSuspenseQuery`: data from the transport is used by the hooks",
+  async () => {
+    let useStaticValueRefStub = <T extends any>(): { current: T } => {
+      throw new Error("Should not be called yet!");
+    };
+    let simulateRequestStart: (options: WatchQueryOptions) => void;
+    let simulateRequestData: (options: Cache.WriteOptions) => void;
+
+    const Provider = WrapApolloProvider(
+      ({ children, onRequestData, onRequestStarted, ...rest }) => {
+        simulateRequestStart = onRequestStarted!;
+        simulateRequestData = onRequestData!;
+        return (
+          <DataTransportContext.Provider
+            value={useMemo(
+              () => ({
+                useStaticValueRef() {
+                  return useStaticValueRefStub();
+                },
+              }),
+              []
+            )}
+          >
+            {children}
+          </DataTransportContext.Provider>
+        );
+      }
+    );
+
+    const client = new WrappedApolloClient({
+      cache: new WrappedInMemoryCache(),
+    });
+
+    let attemptedRenderCount = 0;
+    let finishedRenderCount = 0;
+
+    function Child() {
+      attemptedRenderCount++;
+      const { data } = useSuspenseQuery(QUERY_ME);
+      finishedRenderCount++;
+      return <>{data.me}</>;
+    }
+
+    const { findByText, rerender } = render(
+      <Provider makeClient={() => client}></Provider>
+    );
+
+    simulateRequestStart!(FIRST_REQUEST);
+    rerender(
+      <Provider makeClient={() => client}>
+        <Suspense fallback={"Fallback"}>
+          <Child />
+        </Suspense>
+      </Provider>
+    );
+
+    assert.ok(attemptedRenderCount > 0);
+    assert.ok(finishedRenderCount == 0);
+    await findByText("Fallback");
+
+    useStaticValueRefStub = () => ({ current: FIRST_HOOK_RESULT as any });
+    simulateRequestData!(FIRST_WRITE);
+
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+
+    await findByText("User");
+
+    assert.ok(attemptedRenderCount > 0);
+    assert.ok(finishedRenderCount > 0);
+  }
```

Comment:
Sorry if this is a naive question, but does the cache play any role here? As in, do we need to check the cache to make sure the result made it in the cache?

Reply:
Doesn't hurt - I've added another assertion :)